models/intermediate/int_quickbooks__general_ledger_date_spine.sql

```diff
@@ -4,7 +4,7 @@ with spine as (
 
     {% if execute %}
     {% set first_date_query %}
-        select  min( transaction_date ) as min_date from {{ ref('quickbooks__general_ledger') }}
+        select  coalesce(min(transaction_date),{{ dbt.dateadd("month", -1, "current_date") }}) as min_date from {{ ref('quickbooks__general_ledger') }}
```

Comment:
Same comment as below. In this case I am casting both as date to ensure the result of the dateadd function results in a date datatype and can ensure there are no coalesce datatype mismatching.
```suggestion
        select  coalesce(min(cast(transaction_date as date)),cast({{ dbt.dateadd("month", -1, "current_date") }} as date)) as min_date from {{ ref('quickbooks__general_ledger') }}
```

Reply:
👍